lib/ui/locations/entries/cloudinary.mjs

```diff
@@ -207,85 +251,123 @@ function imageLoad(e, entry) {
+
+    const data = new FormData()
+
+    data.append('file', canvas.toDataURL('image/jpeg', 0.5))
+
+    const response = await fetch(signedUrl, {
+      method: 'post',
+      body: data
     })
 
     if (!response || response.error) {
-      const errorDetail = response?.error?.message ? ` Error: ${response.error.message}` : '';
-      const errorMessage = `${mapp.dictionary.image_upload_failed}${errorDetail}`;
-      alert(errorMessage);
+      const errorDetail = response?.error?.message? `Error: ${response.error.message}` : '';
+      const errorMessage = `Cloudinary Image upload failed! ${errorDetail}`;
+      mapp.ui.elements.alert({text: errorMessage});
       return;
     }
 
+    const responseJson = await response.json()
+
     if (entry.type === 'image') {
 
       // Only a single image is supported by the entry.type.
-      entry.value = response.secure_url
+      entry.value = responseJson.secure_url
+
     } else {
 
       // Add the secure_url to the entry values array and update the location.
-      entry.value = Array.isArray(entry.value) ? entry.value.concat([response.secure_url]) : [response.secure_url]
+      entry.value = Array.isArray(entry.value) ? entry.value.concat([responseJson.secure_url]) : [responseJson.secure_url]
     }
 
-    postUpdate(entry)
+    updateLocation(entry)
   }
 
   img.src = e.target.result
 }
 
+/**
+@function docLoad
+
+@description
+The docLoad method handles the upload of document files to via signed URL to cloudinary.
+
+@param {event} e 
+@param {Object} entry 
+*/
 async function docLoad(e, entry) {
 
-  const response = await mapp.utils.xhr({
-    method: 'POST',
-    requestHeader: {
-      'Content-Type': 'application/octet-stream'
-    },
-    url: `${entry.location.layer.mapview.host}/api/provider/cloudinary?${mapp.utils.paramString({
-      public_id: entry.file.name,
-      resource_type: 'raw',
-      folder: entry.cloudinary_folder
-    })}`,
-    body: e.target.result
+  const date = new Date();
+  const stamp = `${date.getMonth() + 1}-${date.getDate()}T${date.getHours()}:${date.getMinutes()}`
+  const file_type = entry.file.name.substring(entry.file.name.lastIndexOf('.'))
+  const public_id = `${entry.file.name.replace(file_type, '')}-${stamp}${file_type}`;
+
+  const signedUrl = await getSignedUrl(entry, { public_id })
+
+  // Failed to generate a signedUrl for the request.
+  if (!signedUrl) return;
+
+  const data = new FormData()
+  
+  data.append('file', e.target.result.toString())
+
+  const response = await fetch(signedUrl, {
+    method: 'post',
+    body: data
   })
 
   if (!response || response.error) {
-    const errorDetail = response?.error?.message ? ` Error: ${response.error.message}` : '';
-    const errorMessage = `${mapp.dictionary.document_upload_failed}${errorDetail}`;
-    alert(errorMessage);
+    const errorDetail = response?.error?.message? `Error: ${response.error.message}` : '';
+
+    alert(`Cloudinary document upload failed! ${errorDetail}`);
```

Comment:
We will also want to make sure that all alerts are now using the core defined alerts in the module!
```suggestion
    alert(`Cloudinary document upload failed! ${errorDetail}`);
```mapp.ui.elements.alert();